src/main/java/com/exasol/projectkeeper/AbstractPluginPomTemplate.java

```diff
@@ -15,14 +15,19 @@
 import org.w3c.dom.Document;
 import org.w3c.dom.Node;
 import org.xml.sax.SAXException;
+import org.xmlunit.builder.DiffBuilder;
+import org.xmlunit.diff.DefaultNodeMatcher;
+import org.xmlunit.diff.Diff;
+import org.xmlunit.diff.ElementSelectors;
 
 /**
  * Abstract basis for maven plugin configuration validation.
  * <p>
  * Create a new plugin validation by adding a template file to {@code src/main/resources/maven_templates/}. Next create
  * an instance of this class and pass the template's name to the super constructor. If you want to enforce more than
- * only the existence of the plugin definition, override {@link #validatePluginConfiguration(Node, RunMode)}. Finally
- * add your class to {@link PomFileTemplateRunner#TEMPLATES}.
+ * only the existence of the plugin definition, override
+ * {@link #validatePluginConfiguration(Node, RunMode, Collection)}. Finally add your class to
+ * {@link PomFileTemplateRunner#TEMPLATES}.
```

Comment:
I added some additional user guide to the README. I would, however like to keep it here, since I think it is a good idea to keep documentation also in javadoc, since, typically people don't read user guides...